MDANSE_GUI/Src/MDANSE_GUI/Tabs/Models/PlotDataModel.py

```
@@ -154,14 +165,26 @@ class PlotDataModel(QStandardItemModel):
     It stores elements and emits them to the ItemVisualiser.
     """
 
+    DEFAULT_JSON_PATH = (
+        PLATFORM.application_directory() / "recent_plot_selection_file.json"
+    )
+    MAX_NUMBER_RECENT_FILES = 10  # maximum number of recent files to store
+    PLACEHOLDER_STRING = "Recently used result files (.mda)"
+
     error = Signal(str)
     all_elements = Signal(object)
+    finished_loading = Signal(int)
+    recent_files_update = Signal(str)
 
     def __init__(self, parent: QObject = None):
         super().__init__(parent=parent)
         self.mutex = QMutex()
         self._nodes = {}
         self._next_number = 0
+        if not self.DEFAULT_JSON_PATH.exists():
```

If it does exist, it silently does nothing? Is this what we want?

---

In the current implementation of the code, the GUI checks the contents of the JSON file every time we select the 'recently used files' menu, but not before. A weird problem (at least on my machine) caused the menu not to appear at all throughout the GUI session if the JSON file was nonexistent on startup. This code is just meant to make sure that a file with this name exists, which should be the case after the user worked with this version of the code once.